modules/telemetry/src/metric_record.cpp

```diff
@@ -0,0 +1,151 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+#include "hephaestus/telemetry/metric_record.h"
+
+#include <cerrno>
+#include <charconv>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <system_error>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <absl/base/thread_annotations.h>
+#include <absl/log/log.h>
+
+#include "hephaestus/concurrency/message_queue_consumer.h"
+#include "hephaestus/telemetry/metric_sink.h"
+
+namespace heph::telemetry {
+
+namespace internal {
+namespace {
+[[nodiscard]] auto stringToInt64(const std::string& str) -> std::optional<int64_t> {
```

Comment:
This looks rather error prone (also only indirectly tested)...why not use `std::stoi` / `std::stol`? if the optional is needed, with a try/catch, e.g.
```cpp
[[nodiscard]] auto stringToInt64(const std::string& str) -> std::optional<int64_t> {
   try { 
      return std::stol(str);
    } catch (...) {
      return std::nullopt;
    }
 }    

```

Reply:
I have moved the function to string utils and added complete unit tests